rusoto/core/src/compression.rs

```diff
@@ -0,0 +1,33 @@
+use std::rc::Rc;
+use std::sync::Arc;
+use crate::signature::SignedRequest;
+
+pub struct CompressionOptions {
+    pub min_payload_size_for_compression: Option<usize>,
```

Comment:
should we add `compression_type`? e.g. `deflate`, `gzip`?

Reply:
Yes, I will add support for different compression types. Also add `libflate::gzip::EncodeOptions` when we reach to a consensus on how to forward from this point.